plugins/entityreference/behavior/OgBehaviorHandler.class.php

```diff
@@ -122,11 +122,7 @@ public function OgMembershipCrud($entity_type, $entity, $field, $instance, $lang
     foreach ($items as $item) {
       $gid = $item['target_id'];
 
-      // Must provide correct state in the event that approval is required.
-      if (empty($item['state']) && $entity_type == 'user' && !og_user_access($group_type, $gid, 'subscribe without approval', $entity) && !og_user_access($group_type, $gid, 'administer group')) {
```

Comment:
Now this is unreachable code, due to `og_user_access($group_type, $gid, 'administer group')`, versus https://github.com/Gizra/og/blob/7.x-2.x/plugins/entityreference/behavior/OgBehaviorHandler.class.php#L98 , that's why I removed. As far as i see, now it became impossible to trigger that branch.

 removed unreachable code in OgBehaviorHandler.class.php . If the $use…

If the user has group admin permission, we should not set new membership to pending. But if he is not an admin, we exit early from the membership Crud. Adapted the test to test admin permission check of the Crud itself.